### PR TITLE
Refactor Splash screen (and other things)

### PR DESCRIPTION
**Pull request by anttimaki**
This PR got quite big and I considered splitting it and squashing some commits, but decided against it. When I started the work, grasping all that was going on was difficult, so I just started cleaning small parts here and there and putting them in separate commits. I feel that squashing or splitting this process will make it difficult for the reviewers to grasp what was done, why it was done, and whether all the changes were intentional. So I think the PR is best reviewed commit by commit, but YMMV. Let me know if you want it split or squashed.

### CODE REVIEW

File: src/model/ThunderstoreMod.ts [hunk 1]
```diff
@@ -16,35 +16,36 @@ export default class ThunderstoreMod extends ThunderstoreVersion implements Reac
...
-    public parseFromThunderstoreData(data: any): ThunderstoreMod {
-        this.setName(data.name);
-        this.setFullName(data.full_name);
-        this.setOwner(data.owner);
-        this.setDateCreated(data.date_created);
-        this.setDateUpdated(data.date_updated);
-        this.setDeprecatedStatus(data.is_deprecated);
-        this.setPinnedStatus(data.is_pinned);
+    public static parseFromThunderstoreData(data: any): ThunderstoreMod {
+        const mod = new ThunderstoreMod();
```
Comment (ebkr) [inline src/model/ThunderstoreMod.ts:20]: I'm going to assume this hasn't had any side-effects, was there any reason for converting this to a static method?

File: src/App.vue [hunk 2]
```diff
@@ -69,9 +64,10 @@ import InstallationRuleApplicator from './r2mm/installing/default_installation_r
 import GenericProfileInstaller from './r2mm/installing/profile_installers/GenericProfileInstaller';
 import ConnectionProviderImpl from './r2mm/connection/ConnectionProviderImpl';
 import ConnectionProvider from './providers/generic/connection/ConnectionProvider';
+import UtilityMixin from './components/mixins/UtilityMixin.vue';
 
 @Component
-export default class App extends Vue {
+export default class App extends mixins(UtilityMixin) {
```
Comment (ebkr) [inline src/App.vue:70]: Had no idea mixins were a thing but I'm happy with this approach

File: src/components/mixins/SplashMixin.vue [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+import ApiResponse from '../../model/api/ApiResponse';
+import GameManager from '../../model/game/GameManager';
+import RequestItem from '../../model/requests/RequestItem';
+import ConnectionProvider from '../../providers/generic/connection/ConnectionProvider';
+import ThunderstorePackages from '../../r2mm/data/ThunderstorePackages';
+import ApiCacheUtils from '../../utils/ApiCacheUtils';
+
+@Component
+export default class SplashMixin extends Vue {
```
Comment (ebkr) [inline src/components/mixins/SplashMixin.vue:13]: What do we gain by making this a mixin?
If the point was to decouple logic from the template, I'd argue that it hasn't really separated any logic from the page, just moved it.

File: src/model/ThunderstoreMod.ts (hunk 1 above)
Comment (anttimaki) [reply src/model/ThunderstoreMod.ts:20]: There's no reasons for the conversions that aren't up to debate. Main reason was that it works nice with [this implementation](https://github.com/ebkr/r2modmanPlus/pull/854/commits/a79a158b0df9599128c49e4e9845392330096973#diff-aaddca80dd3a9609301da14f24a640a5cc268f2de2f6155e5860127ea4d01252L35-R30), which is the only place where the method is used. Also it was my impression that the static approach is favored in the codebase (e.g. [ExportMod](https://github.com/ebkr/r2modmanPlus/blob/develop/src/model/exports/ExportMod.ts)), but now that I think of it, the impression might stem from the static methods in the singleton classes.

File: src/components/mixins/SplashMixin.vue (hunk 3 above)
Comment (anttimaki) [reply src/components/mixins/SplashMixin.vue:13]: This change was mainly for reducing duplication between r2mm and tsmm, and hopefully making maintaining both easier in the future. It does clean up the [tsmm side](https://github.com/thunderstore-io/thunderstore-mod-manager/pull/109/commits/902574c972b99993de38456f2d4a5b78fce9adc1) quite nicely IMO.

File: src/model/ThunderstoreMod.ts (hunk 1 above)
Comment (ebkr) [reply src/model/ThunderstoreMod.ts:20]: Yeah that's fair enough, happy for this to be kept static

**Review by ebkr — APPROVED** [review]

**Comment by anttimaki** [comment]
Force pushed to fix merge conflicts. Basically merged [my original changes here](https://github.com/ebkr/r2modmanPlus/pull/854/commits/6a8e28c9e37eb1e853040203a5ab8d5567a5ec8a#diff-aaddca80dd3a9609301da14f24a640a5cc268f2de2f6155e5860127ea4d01252L36-R31) with [ebkr's changes here](https://github.com/ebkr/r2modmanPlus/commit/f9c77075375fcac2cd18f3aa5ea2ebd953b35010#diff-aaddca80dd3a9609301da14f24a640a5cc268f2de2f6155e5860127ea4d01252R47-R51).

**Pull request merged**